opensrp-tasking/src/test/java/org/smartregister/tasking/view/TaskingHomeActivityTest.java

```diff
@@ -0,0 +1,865 @@
+import static org.smartregister.tasking.util.TaskingConstants.RequestCode.REQUEST_CODE_FILTER_TASKS;
+import static org.smartregister.tasking.util.TaskingConstants.RequestCode.REQUEST_CODE_GET_JSON;
+import static org.smartregister.tasking.util.TaskingConstants.RequestCode.REQUEST_CODE_TASK_LISTS;
+
+/**
+ * Created by samuelgithengi on 1/23/20.
+ */
+public class TaskingHomeActivityTest extends BaseUnitTest {
+
+    @Rule
+    public MockitoRule rule = MockitoJUnit.rule();
+
+    private TaskingHomeActivity taskingHomeActivity;
+
+    private android.content.Context context = RuntimeEnvironment.application;
+
+    private ImageButton myLocationButton = new ImageButton(context);
+
+    private ImageButton layerSwitcherFab = new ImageButton(context);
+
+    @Mock
+    private TaskingHomePresenter taskingHomePresenter;
+
+    @Mock
+    private TaskingMapHelper taskingMapHelper;
+
+    @Mock
+    private BaseDrawerContract.View drawerView;
+
+    @Mock
+    private Feature feature;
+
+    @Mock
+    private TaskingMapView kujakuMapView;
+
+    @Mock
+    private GeoJsonSource geoJsonSource;
+
+    @Mock
+    private CardDetailsUtil cardDetailsUtil;
+
+    @Mock
+    private MapboxMap mMapboxMap;
+
+    @Mock
+    private ValidateUserLocationPresenter locationPresenter;
+
+    @Mock
+    private ILocationClient locationClient;
+
+    @Mock
+    private Projection projection;
+
+    @Captor
+    private ArgumentCaptor<BoundaryLayer> boundaryLayerArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<CameraPosition> cameraPositionArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<ScaleBarWidget> scaleBarWidgetArgumentCaptor;
+
+    private TaskingLibraryConfiguration taskingLibraryConfiguration;
+
+    @Mock
+    private ValidateAssignmentReceiver validateAssignmentReceiver;
+
+    @Before
+    public void setUp() {
+        Context.bindtypes = new ArrayList<>();
+        taskingHomeActivity = Robolectric.buildActivity(TaskingHomeActivity.class).create().get();
+        FrameLayout.LayoutParams params = new FrameLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);
+        params.topMargin = 30;
+        myLocationButton.setLayoutParams(params);
+        Whitebox.setInternalState(taskingHomeActivity, "myLocationButton", myLocationButton);
+        Whitebox.setInternalState(taskingHomeActivity, "cardDetailsUtil", cardDetailsUtil);
+
+        taskingLibraryConfiguration = spy(TaskingLibrary.getInstance().getTaskingLibraryConfiguration());
+        Whitebox.setInternalState(taskingHomeActivity, "taskingLibraryConfiguration", taskingLibraryConfiguration);
+        ReflectionHelpers.setStaticField(ValidateAssignmentReceiver.class, "instance", validateAssignmentReceiver);
+    }
+
+
+    @Test
+    public void testOnCreate() {
+        assertNotNull(Robolectric.buildActivity(TaskingHomeActivity.class).create().get());
+    }
+
+
+//    @Test
```

Comment:
I am not sure of the intention when commenting these out. Could you do something about it? Create an issue, todo, remove, fix